scripts/FormatSource.ps1

```diff
@@ -0,0 +1,293 @@
+<#
```

Comment:
I don't think we need this script -- couldn't the pre-commit hook could just look something like this?

```sh
for file in `git diff-index --cached --name-only HEAD | grep -iE '\.(cpp|cc|h|hpp)$' ` ; do
  # filter above of course should be updated
  clang-format -n "${file}"
  if [ $? -ne 0 ]; then
    exit $?
  fi
done
```

Reply:
It's an interesting idea. but I defer this change. scripts/FormatSource.ps1 is well tested in MixedReality project.
For example, how to locate the clang-format, how to exclude the iOS folder.

Reply:
scripts/hooks/pre-commit and scripts/hooks/clangFormatFunc are provided as you described.
scripts/hooks/clangFormatFunc defined the changing part.